tc-health-client/build/trafficcontrol-health-client.spec

```diff
@@ -0,0 +1,133 @@
+Requires: git 
+
+%description
+Installs Traffic Control cache health client. See the `tc-health-client` application.
+
+%prep
+tar xvf %{SOURCE0} -C $RPM_SOURCE_DIR
+
+
+%build
+set -o nounset
+# copy license
+cp ${TC_DIR}/LICENSE %{_builddir}
+
+hcdir="tc-health-client"
+hcpath="src/github.com/apache/trafficcontrol/${hcdir}/"
+was_active="unk"
+
+%pre
+
+if [[ -f /etc/trafficcontrol/tc-health-client.json ]]; then
+  active=`systemctl status tc-health-client | awk '/Active:/ {print $2}'`
+  if [[ ${active} = "active" ]]; then
+    ${was_active}="yes"
+    systemctl stop tc-health-client
+  fi
+fi
+
+%install
+hcdir="tc-health-client"
+installdir="/usr/bin"
+mandir="/usr/share/man"
+man1dir="man1"
+
+mkdir -p ${RPM_BUILD_ROOT}/${installdir}
+mkdir -p ${RPM_BUILD_ROOT}/etc/logrotate.d
+mkdir -p ${RPM_BUILD_ROOT}/${mandir}/${man1dir}
+mkdir -p ${RPM_BUILD_ROOT}/etc/trafficcontrol
+mkdir -p ${RPM_BUILD_ROOT}/usr/lib/systemd/system
+
+cp -p ${RPM_SOURCE_DIR}/trafficcontrol-health-client-%{version}/tc-health-client.logrotate ${RPM_BUILD_ROOT}/etc/logrotate.d/tc-health-client.logrotate
+
+src="trafficcontrol-health-client-%{version}"
+cp -p ${RPM_SOURCE_DIR}/${src}/tc-health-client ${RPM_BUILD_ROOT}/${installdir}
+cp -p ${RPM_SOURCE_DIR}/${src}/tc-health-client.json.sample ${RPM_BUILD_ROOT}/etc/trafficcontrol
+cp -p ${RPM_SOURCE_DIR}/${src}/tc-health-client.logrotate ${RPM_BUILD_ROOT}/etc/logrotate.d
+cp -p ${RPM_SOURCE_DIR}/${src}/tc-health-client.service ${RPM_BUILD_ROOT}/usr/lib/systemd/system
+gzip -c -9 ${RPM_SOURCE_DIR}/${src}/tc-health-client.1 > ${RPM_BUILD_ROOT}/${mandir}/${man1dir}/tc-health-client.1.gz
+
+ls ${RPM_BUILD_ROOT}/${mandir}/${man1dir}/
+
+%clean
+rm -rf ${RPM_BUILD_ROOT}
+
+%post
+
+# we want all the cache logs under /var/log/trafficcontrol
+if [[ ! -d /var/log/trafficcontrol ]]; then
+  mkdir -p /var/log/trafficcontrol
+  touch /var/log/trafficcontrol/tc-health-client.log
+fi
+
+if [[ ${was_active} = "yes" ]]; then
+  systemctl start tc-health-client
+fi
+
+# update mandb to put man pages in the whatis database, so apps like 'whatis' and 'apropos' get the new pages
+mandb_out="$(mandb 2>&1)"
+mandb_ret=$?
+if [ $mandb_ret -eq 0 ]; then
+	printf "%s\n" "Updated mandb"
+else
+	printf "Failed to update mandb: code %s\n%s\n" ${mandb_ret} ${mandb_out}
+fi
+
+%preun
+
+if [[ -f /etc/trafficcontrol/tc-health-client.json ]]; then
+  active=`systemctl status tc-health-client | awk '/Active:/ {print $2}'`
+  if [[ ${active} = "active" ]]; then
+    systemctl stop tc-health-client
+  fi
+fi
+
+%postun
+
+# update whatis database, to remove tc-health-client data
+mandb_out="$(mandb 2>&1)"
+mandb_ret=$?
+if [ $mandb_ret -eq 0 ]; then
+	printf "%s\n" "Updated mandb"
+else
+	printf "Failed to update mandb: code %s\n%s\n" ${mandb_ret} ${mandb_out}
+fi
+
+%files
+%license LICENSE
+%attr(755, root, root)
+/usr/bin/tc-health-client
+/usr/share/man/man1/tc-health-client.1.gz
+/etc/trafficcontrol/tc-health-client.json.sample
```

Comment:
Should `/etc/trafficcontrol/tc-health-client.json.sample` be `/etc/trafficcontrol/tc-health-client/config.json.sample` since the config is not shared with other packages using the `/etc/trafficcontrol` directory?

Reply:
@zrhoffman No, I don't see any value in this.  /etc/trafficcontrol/tc-health-client.json.sample is perfectly fine.  When deployed the file will be edited and renamed to just tc-health-client.json.  The name IMHO completely expresses what it's for.  I see no goo reason to create another directory

Reply:
@ocket8888 That seems quite reasonable.  I'll make the change once I know how to proceed with the rest.

Reply:
> @zrhoffman No, I don't see any value in this. /etc/trafficcontrol/tc-health-client.json.sample is perfectly fine. When deployed the file will be edited and renamed to just tc-health-client.json. The name IMHO completely expresses what it's for. I see no good reason to create another directory

Sounds good, thanks for explaining.